test/SlowTests/Issues/RavenDB-14541.cs

```diff
@@ -0,0 +1,247 @@
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_14541 : RavenTestBase
+{
+    public RavenDB_14541(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    private class Address
+    {
+        public string CountryState { get; set; }
+        public string City { get; set; }
+        public string StateId;
+    }
+
+    private class State
+    {
+        public string Name { get; set; }
+    }
+
+    private class QueryResult
+    {
+        public string Comapny { get; set; }
+    }
+
+    private class User
+    {
+        public string UserName { get; set; }
+    }
+
+    [Fact]
+    public void ShouldThrowNotSupportedExceptionHaveIncludesInsteadOf_()
+    {
+        try
+        {
+            using (DocumentStore store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Order { Company = "1", Employee = "employees/1" }, "orders/1-A");
+                    session.Store(new Order { Company = "2", Employee = "employees/2" }, "orders/2-A");
+                    
+                    session.Store(new Employee { FirstName = "a" }, "employees/1");
+                    session.Store(new Employee { FirstName = "b" }, "employees/2");
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query3 = from u in session.Query<Order>()
+                        let includes = RavenQuery.Include<Order>(u.Employee)
+                        select new QueryResult { Comapny = u.Company };
+
+                    var results = query3.ToList();
+                }
+            }
+        }
+        catch (Exception e)
+        {
+            Assert.NotNull(e);
+            Assert.Contains("You can't use the include that way try: let _ = RavenQuery.Include<T>(Expression<Func<T, string>> path)", e.InnerException.Message);
+        }
+    }
+
+    [Fact]
+    public void ShouldThrowNotSupportedExceptionHaveIncorrectObjectAsArg()
+    {
+        try
+        {
+            using (DocumentStore store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                { 
+                    session.Store(new Address { CountryState = "states/1#zip07", City = "new-york", StateId = "states/1" });
+                    session.Store(new Address { CountryState = "states/2#zip05", City = "haifa", StateId = "states/2" });
+
+                    session.Store(new State { Name = "Alabama" }, "states/1");
+                    session.Store(new State { Name = "Minassota" }, "states/2");
+
+                    session.SaveChanges();
+                }
+                using (var session = store.OpenSession())
+                {
+                    var query3 = from a in session.Query<Address>()
+                        let _ = RavenQuery.Include<Address>(a.StateId)
```

Comment:
this is the path case ? I think the correct syntax would be this?
`let _ = RavenQuery.Include<Address>(a => a.StateId)`

also we should forbid such syntax, and allow either string or lambda

Reply:
This is a test that shows we don't allow such syntax; it'll throw and then have an error message with the following:
`You can't use the include that way try: let _ = RavenQuery.Include<T>(Expression<Func<T, string>> path)`

Reply:
its not really testing that it throws, you should wrap the code that should throw in Assert.Throws, something like this:

https://github.com/ravendb/ravendb/blob/de8a154c3ebf19f2ad1faf6e1e97b2024cac2eb3/test/EmbeddedTests/RuntimeFrameworkVersionMatcherTests.cs#L86